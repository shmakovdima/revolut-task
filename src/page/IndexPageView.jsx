import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '../components/Button';
import InfoBlock from '../components/InfoBlock';
import ExchangeItem from '../components/ExchangeItem';

const IndexPageView = ({ startExchange, disabled }) => (
  <PageLayout>
    <ExchangeBlock>
      <LayoutContainer>
        <ExchangeItem
          type="from"
        />
        <InfoBlock />
      </LayoutContainer>
    </ExchangeBlock>
    <ExchangeBlock grey>
      <LayoutContainer>
        <ExchangeItem type="to" />
        <StyledButton
          disabled={disabled}
          onClick={startExchange}
        >
          Exchange
        </StyledButton>
      </LayoutContainer>
    </ExchangeBlock>
  </PageLayout>
);

IndexPageView.propTypes = {
  startExchange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
};

const StyledButton = styled(Button)`
  margin-top: 50px;
  margin-bottom: -20px;
`;

const PageLayout = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
`;

const ExchangeBlock = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 50%;
  padding: 0 20px;
  background: ${props => props.grey && props.theme.grey};
`;

const LayoutContainer = styled.div`
  width: 100%;
  max-width: 600px;
  position: relative;
  height: 100%;
  padding: 10px;
  flex-direction: column;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default React.memo(IndexPageView);
