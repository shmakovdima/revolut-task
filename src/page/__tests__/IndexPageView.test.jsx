import React from 'react';
import { shallow } from 'enzyme';
import IndexPageView from '../IndexPageView';

const props = {
  disabled: false,
  startExchange: jest.fn(),
};

describe('Component [IndexPageView]', () => {
  test('Basic render', () => {
    const component = shallow(<IndexPageView
      {...props}
    />);
    expect(component).toMatchSnapshot();
  });
});
