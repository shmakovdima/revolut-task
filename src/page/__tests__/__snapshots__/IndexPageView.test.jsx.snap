// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component [IndexPageView] Basic render 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Connect(Component)
        type="from"
      />
      <Connect(Component) />
    </styled.div>
  </styled.div>
  <styled.div
    grey={true}
  >
    <styled.div>
      <Connect(Component)
        type="to"
      />
      <Styled(Component)
        disabled={false}
        onClick={[MockFunction]}
      >
        Exchange
      </Styled(Component)>
    </styled.div>
  </styled.div>
</styled.div>
`;
